packages/app/components/drop/drop-form.tsx

```diff
@@ -147,6 +150,28 @@ export const DropForm = () => {
     );
   }
 
+  if (
+    !userProfile?.data.profile.username ||
+    userHasIncompleteExternalLinks(userProfile?.data.profile) ||
+    !userProfile?.data.profile.bio ||
+    !userProfile?.data.profile.img_url
+  ) {
+    return (
+      <View tw="flex-1 items-center justify-center px-10 text-center">
```

Comment:
it would be better if encapsulate this component, we might use it in more places.

Reply:
yeah, intentionally kept it like this. I don't like doing early abstractions 😂 . Will move it once we finalise if we're keeping this functionality.